File: python/test_secret_recipe_decoder.py
from secret_recipe_decoder import decode_ingredient, decode_string, Ingredient

def test_decode_string_can_decode():
    assert(decode_string("yhv") == "abc") # swapped the arg and output

def test_decode_ingredient_can_decode():
    expected = Ingredient("1 cup", "butter")
    actual = decode_ingredient("8 vgl#hgiikf")
    assert(actual.amount == expected.amount)
    assert(actual.description == expected.description)

File: java/src/test/java/com/labs1904/TestSecretRecipeDecoder.java
package com.labs1904;

import org.junit.Assert;
import org.junit.Test;

public class TestSecretRecipeDecoder {
    @Test
    public void testDecodeString() {
        Assert.assertEquals("1 cup", SecretRecipeDecoder.decodeString("8 vgl"));
    }

    @Test
    public void testDecodeIngredient() {
        Ingredient expected = new Ingredient("1 cup", "butter");
        Ingredient actual = SecretRecipeDecoder.decodeIngredient("8 vgl#hgiikf");
        Assert.assertEquals(expected.getAmount(), actual.getAmount());
        Assert.assertEquals(expected.getDescription(), actual.getDescription());
    }
}
